pkgs/by-name/aw/aw-watcher-window-wayland/package.nix

```
@@ -4,36 +4,47 @@
   rustPlatform,
   pkg-config,
   openssl,
+  stdenv,
+  darwin,
+  wayland,
+  unstableGitUpdater,
 }:
 rustPlatform.buildRustPackage rec {
   pname = "aw-watcher-window-wayland";
-  version = "6108ad3df8e157965a43566fa35cdaf144b1c51b";
+  version = "unstable-2024-10-08";
 
   src = fetchFromGitHub {
     owner = "ActivityWatch";
-    repo = pname;
-    rev = version;
-    hash = "sha256-xl9+k6xJp5/t1QPOYfnBLyYprhhrzjzByDKkT3dtVVQ=";
+    repo = "aw-watcher-window-wayland";
+    rev = "58bf86a6984cb01fa750c84ce468c7ccb167f796";
+    hash = "sha256-SnlShM44jnQiZGg5mjreZg1bsjFLNYMjC/krR1TXTE4=";
   };
 
-  cargoPatches = [ ./rustc-serialize-fix.patch ];
-
-  cargoLock = {
-    lockFile = ./Cargo.lock;
-    outputHashes = {
-      "aw-client-rust-0.1.0" = "sha256-9tlVesnBeTlazKE2UAq6dzivjo42DT7p7XMuWXHHlnU=";
-    };
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit pname version src;
+    hash = "sha256-WWT8tOrHPf5x3bXsVPt32VKut4qK+K8gickBfEc0zmk=";
   };
+
+  passthru.updateScript = unstableGitUpdater { };
+
   nativeBuildInputs = [
     pkg-config
   ];
 
-  buildInputs = [
-    openssl
-  ];
+  buildInputs =
+    [
+      openssl
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+      darwin.apple_sdk.frameworks.SystemConfiguration
+    ]
+    ++ lib.optionals stdenv.isLinux [
+      wayland
+    ];
 
   meta = with lib; {
```

```suggestion
  meta = {
```

Nit: since you're refactoring most of the package already, might as well remove this nested `with lib;` statement. Not a blocker though.